sympy/integrals/tests/test_meijerint.py

```diff
@@ -725,3 +726,10 @@ def test_issue_6462():
     # exception
     assert integrate(cos(x**n)/x**n, x, meijerg=True).subs(n, 2).equals(
             integrate(cos(x**2)/x**2, x, meijerg=True))
+
+
+def test_issue_21063():
+    from sympy import Symbol
+    x = Symbol('x')
+    assert integrate(exp(-x ** 2), (x, -oo, oo)) == sqrt(pi)
+    assert integrate(exp(-x ** 2), (x, Mul((-1), oo, evaluate=False), oo)) == sqrt(pi)
```

Comment:
There should be tests covering all paths that you have changed.

Reply:
Thanks, for reviewing the pr. I am not sure I understand what you mean by this, can you please elaborate?

Reply:
You have made changes for several different cases where the unevaluated Mul can appear in one or both limits. There should be tests for each case.